### DISCUSSION TITLE
Unable to get CORS Middleware to work with Github Codespaces despite following docs exactly and checking other discussions.

### DISCUSSION THREAD

**Discussion by PhantomLel** [fastapi/fastapi#10246] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()
origins = [
    "http://localhost:5173", "http://localhost:8000", "https://<Github CodeSpaces URL>"
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["POST", "GET", "PUT", "DELETE", "OPTIONS", "HEAD", "PATCH"],
    allow_headers=["Authorization", "Content-Type", "Accept"],
    expose_headers=["*"]
)
```

### Description

I am using Github Codespaces to develop an app and I am trying to connect the frontend to the backend. It keeps giving me a CORS error despite setting up the middleware exactly how the docs say. I tried putting both the localhost url and the actual codespaces url but neither works. I also tried importing the middleware from Starlette. I think the error might have something to do with codespaces but I have no idea how to fix it.

### Operating System

Linux

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.10.8

### Additional Context

** Edit

I also forgot to show how I am requesting from the frontend:
```js
  fetch(urls.auth.login, {
      method: "POST",
      headers: {
          "content-type": "application/x-www-form-urlencoded",
      },
      body: <data>
  })```

**Comment by Leek727** [comment]
DUDE, I HAVE THE SAME PROBLEM!

**Comment by PhantomLel** [comment, marked as answer]
I seem to have figured something out. For some reason, when I request `127.0.0.1` it works but when I use `localhost` it does not.
